sdk/src/shortvec.rs

```diff
@@ -1,373 +1,37 @@
-use bincode::{deserialize_from, serialize_into, Error};
+use serde::de::Deserializer;
+use serde::ser::Serializer;
 use serde::Serialize;
-use std::io::{Cursor, Read, Write};
-use std::mem::size_of;
 
-pub fn encode_len<W: Write>(writer: &mut W, len: usize) -> Result<(), Error> {
-    let mut rem_len = len;
+// Use 1 to 9 bytes to store the length of the vector.
+pub fn serialize<S: Serializer, T: Serialize>(
+    elements: &[T],
+    serializer: S,
+) -> Result<S::Ok, S::Error> {
+    use serde::ser::SerializeTuple;
+    let mut seq = serializer.serialize_tuple(0)?;
+
+    let mut rem_len = elements.len();
     loop {
         let mut elem = (rem_len & 0x7f) as u8;
         rem_len >>= 7;
         if rem_len == 0 {
-            writer.write_all(&[elem])?;
+            seq.serialize_element(&elem)?;
```

Comment:
@sambley Using the same tuple for both the multi-byte length and the elements in the vector might be a terrible idea.  A struct of two tuples might work better.